src/cbmc/symex_bmc.cpp

```diff
@@ -131,25 +131,12 @@ bool symex_bmct::get_unwind(
   // / --unwind options to decide:
   if(abort_unwind_decision.is_unknown())
   {
-    // We use the most specific limit we have,
-    // and 'infinity' when we have none.
+    auto limit=unwindset.get_limit(id, source.thread_nr);
 
-    loop_limitst &this_thread_limits=
-      thread_loop_limits[source.thread_nr];
-
-    loop_limitst::const_iterator l_it=this_thread_limits.find(id);
-    if(l_it!=this_thread_limits.end())
-      this_loop_limit=l_it->second;
+    if(limit.has_value())
+      abort_unwind_decision = tvt(false);
     else
-    {
-      l_it=loop_limits.find(id);
-      if(l_it!=loop_limits.end())
-        this_loop_limit=l_it->second;
-      else if(max_unwind_is_set)
-        this_loop_limit=max_unwind;
-    }
-
-    abort_unwind_decision = tvt(unwind >= this_loop_limit);
+      abort_unwind_decision = tvt(unwind >= *limit);
```

Comment:
Since the very same code exists twice: how about adding a `tvt unwindsett::limit_reached(id, thread_nr, unwind)`?

Reply:
(I want to get rid of the handlers, BTW, that is the whole point of overloading those methods in goto-symex)

Reply:
I don't really understand. Lines 132...137 are, except for `source.thread_nr` vs `thread_nr`, exactly the same as lines 175..180.

Reply:
Which, btw, changes the behaviour: in the previous code, one used `>=` in the comparison, while the other used `>`.  (I'm not sure they actually should be different...)